server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -351,27 +353,48 @@ public void trimUnreferencedReaders() throws IOException {
             generationsToDelete.add(generation);
         }
         if (generationsToDelete.isEmpty() == false) {
-            deleteRemoteGenerationAsync(generationsToDelete);
-            deleteOlderPrimaryTranslogFiles();
+            deleteRemoteGeneration(generationsToDelete);
```

Comment:
This should take care of deleting the metadata as well . I would prefer that as part of `TranslogTransferManager#deleteTranslogAsync` . This is as metadata is a part of one generation upload .

Reply:
I see another benefit of one lesser call. So, incorporated.